src/dials/array_family/flex_ext.py

```diff
@@ -375,6 +378,110 @@ def as_hdf5(self, filename):
+        ds.coords["vec2_index"] = [0, 1]
+        ds.coords["vec3_index"] = [0, 1, 2]
+        ds.coords["int6_index"] = [0, 1, 2, 3, 4, 5]
+        ds.coords["mat3_index1"] = [0, 1, 2]
+        ds.coords["mat3_index2"] = [0, 1, 2]
+        for key, data in self.cols():
+            if ignore and key in ignore:
+                continue
+            if isinstance(data, dials_array_family_flex_ext.shoebox):
+                d = shoebox_to_dict(data)
+                ds["bbox"] = ((index_name, "int6_index"), d["bbox"].reshape(sz, 6))
+                ds["panel"] = ((index_name), d["panel"])
+                # i = 0
+                # # TODO Since xarray requires named axes and coordinates, an arbitrary-shape shoebox is not allowed.
+                # # Bboxes and panels are allowed, but not sbdata, bg, or mask (unless they are guaranteed to be of a fixed shape for the entire reflection table.)
+                # for bbox in d["bbox"]:
+                #     vol = (
+                #         (bbox[5] - bbox[4]) * (bbox[3] - bbox[2]) * (bbox[1] - bbox[0])
+                #     )
+                #     sbdata = np.reshape(
+                #         d["shoebox_data"][i : i + vol],
+                #         ((bbox[5] - bbox[4]), (bbox[3] - bbox[2]), (bbox[1] - bbox[0])),
+                #     )
+                #     bg = np.reshape(
+                #         d["shoebox_background"][i : i + vol],
+                #         ((bbox[5] - bbox[4]), (bbox[3] - bbox[2]), (bbox[1] - bbox[0])),
+                #     )
+                #     mask = np.reshape(
+                #         d["shoebox_mask"][i : i + vol],
+                #         ((bbox[5] - bbox[4]), (bbox[3] - bbox[2]), (bbox[1] - bbox[0])),
+                #     )
+                #     print(sbdata)
+                #     print(bg)
+                #     print(mask)
+                #     i += vol
+            elif isinstance(data, dials_array_family_flex_ext.int6):
+                ds[key] = (
+                    (index_name, "int6_index"),
+                    flumpy.to_numpy(data.as_int()).reshape(sz, 6),
+                )
+            elif isinstance(data, cctbx.array_family.flex.vec2_double):
+                ds[key] = (
+                    (index_name, "vec2_index"),
+                    flumpy.to_numpy(data).reshape(sz, 2),
+                )
+            elif isinstance(data, cctbx.array_family.flex.vec3_double):
+                ds[key] = (
+                    (index_name, "vec3_index"),
+                    flumpy.to_numpy(data).reshape(sz, 3),
+                )
+            elif isinstance(data, cctbx.array_family.flex.mat3_double):
+                ds[key] = (
+                    (index_name, "mat3_index1", "mat3_index2"),
+                    flumpy.to_numpy(data).reshape(sz, 3, 3),
+                )
+            elif isinstance(data, cctbx.array_family.flex.std_string):
+                ds[key] = ((index_name), list(data))
```

Comment:
Is this for completion' sake or is there a usage of this that you found? My memory was that array_family string objects were _completely_ broken..